massdriver-application-azure-app-service/_variables.tf

```diff
@@ -44,10 +44,7 @@ variable "virtual_network_id" {
 }
 
 variable "network" {
-  type = object({
-    auto = bool
-    cidr = string
-  })
+  type = any
```

Comment:
Should this be a string and allow the variable to be optional? If its null then auto mode, else, take the value?

Reply:
Yes. @WillBeebe mentioned it wouldn't be possible unless we upgraded our Terraform provisioner to v1.3+

Reply:
The string would be required and we could pass a blank `""` but that felt icky. 

TF `1.3.0` introduced `optional(string)` which is what we all want.

Reply:
What's fun is that the downstream TF you write with `any` or `optional` is still the same. 

With an `any` type you can have an optional and thus a `null` string attribute. This lets us do things like `count = var.optional_thing  == null ? 0 : 1`. 

I put together a PR that tests building our Terraform provisioner with a newer version.

https://github.com/massdriver-cloud/massdriver-bundles/pull/384

Reply:
@chrisghill ^